### DISCUSSION TITLE
New JQuery version for modern browsers

### DISCUSSION THREAD

**Discussion by luz-arreola** [jquery/jquery#5101] [Ideas]
People love hating on JQuery and the two main reasons hear are that 1) everything can be done using vanilla JS and 2) library size.  They love to say that JQuery is no longer needed because browser compatibility issues are gone and that JQuery is no longer needed.

I'm a backend developer and JQuery makes working with the DOM so much simpler and code more concise.  I would imagine that dedicated frontend JQuery users would agree.

I really do believe that there should be a newer, very lightweight version that strips out all browser compatibility code.  It will give the haters one less thing to point out.

**Comment by mgol** [comment]
Thanks for the kind words! Let me convert this into a discussion.

We’re planning to release a beta for the new major version, 4.0.0, soon (I can’t provide exact timelines but we’re getting close). This version drops some of the older browsers, you can see the exact difference in this WIP PR to our browser support docs: https://github.com/jquery/jquery.com/pull/194

The most important changes there are dropping support for IE 9-10, older iOS versions, Edge Legacy and the old Android Browser. We’ve also inlined Sizzle, the jQuery selector engine and removed lots of workarounds for these & even older browsers.

We had bigger plans for jQuery 4.0.0 but we’ve decided it’s time to do a release after all these years so we pushed some changes to 5.0.0. That includes dropping support for IE 11 - jQuery 4.x will be the last line to support any version of IE.

I hope that answers your questions!

**Reply by Tommybio** [reply]
Thank you

**Reply by luz-arreola** [reply]
Michal, thank you so much for the exciting news!  I'm so glad to hear!

On another note, in C++, you can make a very small executable or a large one, and this depends on **what you include** into your project.

Just my most humble thought...having an online bundler allowing you to pick and choose desired features.  For example, including/excluding deprecated features, animation effects, etc.

One last comment that I'd like to make is that backend developers can obtain the client browser version and modify the markup to send the client.  Depending on the client, they will bring in either the new version of JQuery (for modern browsers), or an older JQuery version if he or she is on an older browsers.

**Reply by mgol** [reply]
An online bundler is in our [roadmap](https://github.com/jquery/jquery/wiki/Roadmap) but there are no concrete plans for now. You can build a custom version of jQuery by yourself by following instructions in README, though: https://github.com/jquery/jquery/blob/main/README.md#modules.

> One last comment that I'd like to make is that backend developers can obtain the client browser version and modify the markup to send the client. Depending on the client, they will bring in either the new version of JQuery (for modern browsers), or an older JQuery version if he or she is on an older browsers.

This is not a good idea. There are differences between major versions of jQuery and loading one or the other can change the behavior in unpredictable ways. Always use a version that supports all the environments that your project supports.

**Reply by luz-arreola** [reply]
Thank you kindly Michal for the useful link and the advice.

**Reply by ShadBalti** [reply]
🎉 **Exciting News! Looking Forward to 4.0.0 and Beyond**

Thanks for the detailed update! It's great to hear about the upcoming beta release for version 4.0.0 and the strategic decisions made to streamline browser support.

The shift away from older browsers and the optimizations, including inlining Sizzle and removing workarounds, sound promising. It's understandable to adapt and prioritize modern environments.

The decision to reserve jQuery 4.x as the last version supporting any iteration of IE aligns with the evolving landscape. Excited to see how these changes enhance performance and maintainability. Looking forward to the beta release and the progression to 5.0.0! 🚀

**Comment by MarcoEvasi** [comment]
I agree with luz-arreola about haters pointing out the library size, and I also thought many times that what is missing is modularity.

Take ajax support: I think jquery still does the best job nowadays.
E
ven without an online bundler if that is expensive to develop, I think it would help to have many compiled versions:
- complete
- only ajax
- ajax + DOM
- ...
so everybody could choose only what she needs reducing the file size.

**Comment by ShadBalti** [comment]
🚀 **Embracing jQuery's Simplicity**

Critics often dismiss jQuery, claiming vanilla JS suffices and highlighting concerns about library size. They argue that improved browser compatibility renders jQuery obsolete.

💻 As a backend developer, I've found jQuery's DOM manipulation to enhance code conciseness. Proposing a newer, ultra-lightweight version sans redundant browser compatibility code could silence critics and cater to those who value jQuery's efficiency.
